pkg/collect/redact.go

```diff
@@ -1,20 +1,75 @@
 package collect
 
 import (
+	"archive/tar"
+	"bytes"
+	"encoding/binary"
+	"io"
+	"path"
+
 	troubleshootv1beta1 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta1"
 	"github.com/replicatedhq/troubleshoot/pkg/redact"
 )
 
 func redactMap(input map[string][]byte, additionalRedactors []*troubleshootv1beta1.Redact) (map[string][]byte, error) {
 	result := make(map[string][]byte)
 	for k, v := range input {
-		if v != nil {
-			redacted, err := redact.Redact(v, k, additionalRedactors)
+		if v == nil {
+			continue
+		}
+		//If the file is a .tar file, it must not be redacted. Instead it is decompressed and each file inside the
+		//tar is decompressed, redacted and compressed back into the tar.
+		if path.Ext(k) == ".tar" {
+			tarFile := bytes.NewBuffer(v)
+			buff := new(bytes.Buffer)
+			unRedacted, fileHeaders, err := untarFile(tarFile)
 			if err != nil {
 				return nil, err
 			}
-			result[k] = redacted
+			files, err := redactMap(unRedacted, additionalRedactors)
+			if err != nil {
+				return nil, err
+			}
+			tw := tar.NewWriter(buff)
```

Comment:
could you make this its own function `tarFiles`?

Reply:
Yes, in fact I was considering that approach too. I chose not to  make a function to save some lines of code, and because I did not had to use it in other part of the code, but I could totally make a function which receives the files and headers, and return a tar file, thinking in using it later too. Do you prefer it that way?